mod/state-transition/pkg/core/state_processor.go

```diff
@@ -219,6 +221,23 @@ func (sp *StateProcessor[
 			return nil, err
 		}
 
+		// Handle special cases
+		if sp.cs.DepositEth1ChainID() == spec.BoonetEth1ChainID &&
+			slot == math.U64(spec.BoonetFork2Height) {
+			var idx uint64
+			idx, err = st.GetEth1DepositIndex()
+			if err != nil {
+				return nil, fmt.Errorf(
+					"failed retrieving deposit index at slot %d: %w",
+					slot, err,
+				)
+			}
+			fixedDepositIdx := idx - 1
+			if err = st.SetEth1DepositIndex(fixedDepositIdx); err != nil {
+				return nil, err
+			}
+		}
```

Comment:
the Eth1DepositIndex was wrongly set to +1 the expected value. We fix this at fork.

Reply:
can you mention this in a comment under line 226?